### PR TITLE
Documented theory of Classical Electrodynamics example - Electric Field and implemented the code

### PR DESCRIPTION
**Pull request by Cosmos491**
The documentation includes the theory aspect related to Electric Field ( example of Classical Electrodynamics) and how along with the Maxwell's equations they describe a great deal pf physics. The python file includes a code which calculates the Electric field due to a single charge and then the net Electric Field due to multiple charges using the superposition principle.

### CODE REVIEW

**Comment by laserlab** [comment]
@abruns123 Please review asap

**Review by abruns123 — CHANGES_REQUESTED** [review]
Code runs fine, albeit it only gets a 6.5 score for linting; however, there's a severe issue with the physics of your simulation. The electric field is a vector, but your code is treating it purely as a magnitude. So, for example if we had two equal charges spaced equally apart from the origin on the same axis, I would expect that the electric field components at the origin would be of equal magnitude and opposite direction such that they cancel out. Your code instead just adds their magnitudes together. With some modification, you could easily get this code to determine the electric potential at a certain point due to point charges, but to get electric field accurately, you need to account for the vectors.

**Review by abruns123 — CHANGES_REQUESTED** [review]
Your Electric_FIeld_due_to_single_charge module doesn't run due to an indent error on line 52

**Comment by Cosmos491** [comment]
Yes, I am working on that. Thank you for letting me know. I believe I will be able to implement the changes before the class tomorrow.

File: electric_field_due_to_single_charge.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    # List comprehension to gather x and y components of electric fields from all charges
+    fields_x = [field[0] for field in fields]
+    fields_y = [field[1] for field in fields]
+
+    # Use sum to sum up all x and y components to get the net field
+    net_field_x = sum(fields_x)
+    net_field_y = sum(fields_y)
+
+    return net_field_x, net_field_y
+# Example
```
Comment (laserlab) [inline electric_field_due_to_single_charge.py:64]: Examples should go into the docstring

File: electric_field_due_to_single_charge.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+due to multiple point charges using Coulomb's law.
+"""
+
+import numpy as np
+
+# Coulomb constant in Nm^2/C^2
+k = 8.99e9
+
+# Lambda functions
+separation_vector = lambda p, o: (p[0] - o[0], p[1] - o[1])  # Separation vector between two points
```
Comment (laserlab) [inline electric_field_due_to_single_charge.py:12]: Agreeing here with the linter:
electric_field_due_to_single_charge.py:12:20: C3001: Lambda expression assigned to a variable. Define a function using the "def" keyword instead. (unnecessary-lambda-assignment)
just use a function here

File: electric_field_due_to_single_charge.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+magnitude = lambda v: np.sqrt(v[0]**2 + v[1]**2)  # Magnitude of a 2D vector
```
Comment (laserlab) [inline electric_field_due_to_single_charge.py:13]: see above

**Review by laserlab — CHANGES_REQUESTED** [review]
- check linting issues
- main loop

**Comment by Cosmos491** [comment]
@laserlab and @abruns123 , I have made the changes as requested.

**Review by laserlab — APPROVED** [review]

**Pull request merged**